### ISSUE TITLE
Include `filter`, `select`, and `exclude` arguments in generate source

### ISSUE THREAD

**Issue by mattppal** [datacoves/dbt-coves#271]
**Describe the solution you'd like**
We're currently able to supply 
- `select`,
- `exclude`,
- `filter`

arguments when generating properties. It would be very helpful to do the same when generating sources.

**Describe alternatives you've considered**
Manually selecting sources via the CLI

**Comment by mattppal** [comment]
@ssassi I just read through #41— that's exactly the same use case I have, I want to exclude `_airbyte_raw` tables. I'd be happy to contribute here, but I'm not too familiar with the codebase or setup. Would you be able to meet and chat through the process? I haven't contributed to an open source project before.

**Comment by BAntonellini** [comment]
@mattppal thank you for contributing.

We've recently added `select-relations` and `exclude-relations` to `generate sources` command. They let you choose or ignore relations (tables) from being subject of generation.

Keep in mind filtering in `generate sources` and `generate properties` works differently:
- generate properties works with `select`, `exclude` and `selector`: this filtering strategies work exactly as [dbt ls](https://docs.getdbt.com/reference/commands/list) do. In fact, as `generate properties` works with dbt's local files, this flags are passed directly to a dbt subprocess.
- generate sources works with the two most recent flags I stated above and, instead of working locally, it goes through your Adapter to the database-schema combination passed to the command.

Regards
